### ISSUE TITLE
Auto-import of nitro utils stopped working in 3.14

### ISSUE THREAD

**Issue by ronenteva** [unjs/unimport#399]
### Environment

```
- Operating System: Linux
- Node Version:     v20.18.0
- Nuxt Version:     3.14.1592
- CLI Version:      3.16.0
- Nitro Version:    2.10.4
- Package Manager:  pnpm@9.14.4
```

### Reproduction

/server/utils/sum.ts
```
export function useSum(a: number, b: number) { return a + b }
```
/server/api/result:
```
export default defineEventHandler(() => {
  const sum = useSum(1, 2)
  return { sum }
})
```

`ERROR  useSum is not defined`

### Describe the bug

Everything works fine when pinning `unimport@3.13`

**Comment by noootwo** [comment]
Please provide your config

**Comment by ronenteva** [comment]
What do you mean?

**Comment by vacnex** [comment]
It has been reported [Issue](https://github.com/nuxt/nuxt/issues/30098)

**Comment by danielroe** [comment]
this seems like it might be a regression from https://github.com/unjs/unimport/pull/395.

**Comment by noootwo** [comment]
The `nitro` passes the glob `server/utils/*`:
https://github.com/nitrojs/nitro/blob/575144f0efa9e8115251abb8d0bd09165ecd55f2/src/core/config/resolvers/imports.ts#L30
In versions `< 3.14.x`, `unimport` would scan the raw `dirs` glob and the glob like `{dir glob}/*.{ts,js,...}`, which worked for `nitro`.
After #395, `unimport` lost the ability to scan the raw `dirs` glob, as it only scanned with the glob `xxx/server/utils/*/*.{ts,js}`, which could not match anything, so the case of `nitro` was not handled.

It will be fixed in issue #400.

**Comment by noootwo** [comment]
> version 3.14.3 seems to work a little better, but I'm currently having issues autoimporting files within a subfolder of `server/utils` in Nuxt. I tried adding various paths manually to `nitro.imports.dirs` in `nuxt.config.js`, to no avail. Maybe I'm doing it wrong? It used to work fine without needing to be specific in version 3.13.

I found that in version 3.13, `nitro` can only automatically import files within two levels of the`server/utils`, this may be  unexpected.
